### PR TITLE
msgviewer: Support for utf8 control char payloads

### LINKED ISSUE #660: Differentiate null and empty string as keys in UI
### Version
```
Kowl CI
v1.5.0-598117c
(built November 10, 2021)
```
Unfortunately I cannot test it on the latest release, maybe it is already fixed, I don't know.

### Problem
Producing messages with `key=null` (no key) and with `key=""` leads to the same view in UI:
image

But `kafka-console-consumer` with `--property print.key=true` differentiates these two situations:
- for null key it is:
```
null    <msg1>
null    <msg2>
...
```
- for "" key it is:
```
        <msg1>
        <msg2>
...
```

This fix could help me a lot in debugging problems with assigning messages to partitions: messages with null key are usually uniformly distributed to partitions but messages with `""` key always go to one partition.

### PR DESCRIPTION
**Pull request by weeco**
Fixes #660

### CODE REVIEW

File: frontend/src/state/restInterfaces.ts [hunk 1]
```diff
@@ -151,6 +151,7 @@ export enum CompressionType {
 
 export interface Payload {
     payload: any, // json obj
+    isPayloadNull: boolean,
```
Comment (rikimaru0345) [inline frontend/src/state/restInterfaces.ts:154]: is this actually getting populated by the backend?

Comment (weeco) [reply frontend/src/state/restInterfaces.ts:154]: Yes, backend code change is also part of this PR

File: frontend/src/state/restInterfaces.ts [hunk 2]
```diff
@@ -171,12 +172,11 @@ export interface TopicMessage {
     key: Payload,
     value: Payload,
 
-    isValueNull: boolean, // todo: rename to isTombstone
-
     // Added by the frontend
     valueJson: string,
     valueBinHexPreview: string,
     keyJson: string,
+    keyBinHexPreview: string,
```
[the remark this answers is not in the document]
Comment (weeco) [reply frontend/src/state/restInterfaces.ts:179]: from here: https://github.com/redpanda-data/console/pull/667/files#diff-b0f2d81a7446ac88a7118694dc5b3935d60118d7b0322facf55f39046920ef61R387

**Review by rikimaru0345 — APPROVED** [review]

File: frontend/src/state/backendApi.ts [hunk 3]
```diff
@@ -405,6 +402,16 @@ const apiStore = {
...
 
+    base64ToHexString(b64: string): string {
+        let hex = '';
+        for (let i = 0; i < b64.length && i < 50; i++) {
+            let n = b64.charCodeAt(i).toString(16);
+            if (n.length == 1) n = '0' + n;
+            hex += n + ' ';
+        }
+        return hex
+    },
```
Comment (rikimaru0345) [inline frontend/src/state/backendApi.ts:413]:
```suggestion
    function base64ToHexString(base64: string): string {
        const binary = atob(base64);
        const bytes = new Uint8Array(binary.length);
        for (let i = 0; i < binary.length; i++) {
            bytes[i] = binary.charCodeAt(i);
        }
        
        let hex = '';
        for (let i = 0; i < bytes.length; i++) {
            const b = bytes[i].toString(16);
            hex += b.length === 1 ? '0' + b : b;
            
            if (i < bytes.length - 1)
                hex += ' ';
        }
        
        return hex;
    },
```

**Pull request merged**